C/common/storage_asset_tracking.cpp

```diff
@@ -169,6 +175,26 @@ void StorageAssetTracker::updateCache(std::set<std::string> dpSet, StorageAssetT
         {
                 Logger::getLogger()->debug("%s:%d :tuple not found in cache ", __FUNCTION__, __LINE__);
 		storageAssetTrackerTuplesCache[ptr] = dpSet;
+
+		std::string strDatapoints;
+                unsigned int count = 0;
+                for (auto itr : dpSet)
+                {
+                        strDatapoints.append(itr);
+                        strDatapoints.append(",");
+                        count++;
+                }
+                if (strDatapoints[strDatapoints.size()-1] == ',')
+                        strDatapoints.pop_back();
+
+	       bool rv = m_mgtClient->addStorageAssetTrackingTuple(ptr->getServiceName(), ptr->getPluginName(), ptr->getAssetName(), ptr->getEventName(), false, strDatapoints, count);
+               if (rv)
+               {
+                       storageAssetTrackerTuplesCache[ptr] = dpSet;
```

Comment:
Where do we handle the case if the cache size gets exceeded?

Reply:
cache memory is dynamic memory  , do you mean to say that what if dynamic memory allocated at start gets exhausted ?

Reply:
It seems the current implementation is ok in this regard. There is no upper limit on cache size.